cairo/src/context.rs

```diff
@@ -393,40 +394,49 @@ impl Context {
         unsafe {
             ffi::cairo_fill_extents(self.0.as_ptr(), &mut x1, &mut y1, &mut x2, &mut y2);
         }
-        (x1, y1, x2, y2)
+        self.status().map(|_| (x1, y1, x2, y2))
     }
 
-    pub fn in_fill(&self, x: f64, y: f64) -> bool {
-        unsafe { ffi::cairo_in_fill(self.0.as_ptr(), x, y).as_bool() }
+    pub fn in_fill(&self, x: f64, y: f64) -> Result<bool, Error> {
```

Comment:
Same question here

Reply:
If you have something like

```rust
cr.move_to(...);
cr.line_to(...);
cr.curve_to(...);
// big, complex path gets formed
// maybe cairo OOMs or hopefully detects out-of-range coordinates or whatever

// The result is meaningless if the cr went into an error state above.
// Also maybe if x/y are out of range, cairo could detect it and return an error.
let hit_test = cr.in_fill(x, y)?;
```

So this one and the following are about catching meaningless results in case your code didn't notice before that the cr had gone into an error state.